packages/perseus/src/widgets/radio/utils/hash-shuffle.ts

```diff
@@ -0,0 +1,106 @@
+import {generateChoiceId} from "@khanacademy/perseus-core";
+
+import type {RadioChoiceWithMetadata} from "../radio.class.new";
+
+/**
+ * MumurHash3 32-bit implementation
+ * Implementation reference: https://mojoauth.com/hashing/murmurhash-in-typescript
+ * Optimized for string inputs without external dependencies
+ */
+function hashString(key: string): number {
+    // djb2 constant
+    const hashConstant = 5381;
+
+    let h1 = hashConstant ^ key.length;
+    const c1 = 0xcc9e2d51;
+    const c2 = 0x1b873593;
+
+    for (let i = 0; i < key.length; i++) {
+        let k1 = key.charCodeAt(i);
+        k1 = k1 * c1;
+        k1 = (k1 << 15) | (k1 >>> 17); // Rotate left
+        k1 = k1 * c2;
+        h1 ^= k1;
+        h1 = (h1 << 13) | (h1 >>> 19); // Rotate left
+        h1 = h1 * 5 + 0xe6546b64; // Mix
+    }
+    h1 ^= key.length;
+    h1 ^= h1 >>> 16; // Finalization
+    h1 = h1 * 0x85ebca6b;
+    h1 ^= h1 >>> 13;
+    h1 = h1 * 0xc2b2ae35;
+    h1 ^= h1 >>> 16;
+    return h1 >>> 0;
+}
+
+/**
+ * Creates a hash-based shuffle key for a choice
+ */
+function createShuffleKey(
+    choice: RadioChoiceWithMetadata,
+    seed: string,
+): string {
+    // Use choice ID if available, otherwise generate one
+    const choiceIdentifier =
+        choice.id || generateChoiceId(choice.content, choice.originalIndex);
```

Comment:
Why would a choice not have an ID? Shouldn't the PerseusItem have gone through the parser already?